Source/WebCore/dom/MouseRelatedEvent.h

```diff
@@ -45,70 +45,77 @@ class MouseRelatedEvent : public UIEventWithKeyState {
+
     bool isSimulated() const { return m_isSimulated; }
     void setIsSimulated(bool value) { m_isSimulated = value; }
-    int pageX() const final;
-    int pageY() const final;
-    WEBCORE_EXPORT FloatPoint locationInRootViewCoordinates() const;
+    double pageX() const override;
+    double pageY() const override;
+    WEBCORE_EXPORT DoublePoint locationInRootViewCoordinates() const;
 
     // Page point in "absolute" coordinates (i.e. post-zoomed, page-relative coords,
     // usable with RenderObject::absoluteToLocal).
-    const LayoutPoint& absoluteLocation() const { return m_absoluteLocation; }
-    
+    const DoublePoint& absoluteLocation() const { return m_absoluteLocation; }
+
     static LocalFrameView* frameViewFromWindowProxy(WindowProxy*);
 
-    static LayoutPoint pagePointToClientPoint(LayoutPoint pagePoint, LocalFrameView*);
-    static LayoutPoint pagePointToAbsolutePoint(LayoutPoint pagePoint, LocalFrameView*);
+    static DoublePoint pagePointToClientPoint(DoublePoint pagePoint, LocalFrameView*);
+    static DoublePoint pagePointToAbsolutePoint(DoublePoint pagePoint, LocalFrameView*);
+
+    WEBCORE_EXPORT virtual double offsetX();
+    WEBCORE_EXPORT virtual double offsetY();
+
+    void computeRelativePosition();
 
 protected:
     MouseRelatedEvent(enum EventInterfaceType);
     MouseRelatedEvent();
     MouseRelatedEvent(enum EventInterfaceType, const AtomString& type, CanBubble, IsCancelable, IsComposed, MonotonicTime, RefPtr<WindowProxy>&&, int detail,
-        const IntPoint& screenLocation, const IntPoint& windowLocation, double movementX, double movementY, OptionSet<Modifier> modifiers,
+        const DoublePoint& screenLocation, const DoublePoint& windowLocation, double movementX, double movementY, OptionSet<Modifier> modifiers,
         IsSimulated = IsSimulated::No, IsTrusted = IsTrusted::Yes);
-    MouseRelatedEvent(enum EventInterfaceType, const AtomString& type, IsCancelable, MonotonicTime, RefPtr<WindowProxy>&&, const IntPoint& globalLocation, OptionSet<Modifier>);
+    MouseRelatedEvent(enum EventInterfaceType, const AtomString& type, IsCancelable, MonotonicTime, RefPtr<WindowProxy>&&, const DoublePoint& globalLocation, OptionSet<Modifier>);
     MouseRelatedEvent(enum EventInterfaceType, const AtomString& type, const MouseRelatedEventInit&, IsTrusted = IsTrusted::No);
 
     void initCoordinates();
-    void initCoordinates(const LayoutPoint& clientLocation);
+    void initCoordinates(const DoublePoint& clientLocation);
     void receivedTarget() override;
 
     void computePageLocation();
-    void computeRelativePosition();
 
     float documentToAbsoluteScaleFactor() const;
 
     // Expose these so MouseEvent::initMouseEvent can set them.
-    IntPoint m_screenLocation;
-    LayoutPoint m_clientLocation;
+    DoublePoint m_screenLocation;
+    DoublePoint m_clientLocation;
+
+    // Expose these so PointerEvent can access it.
+    // FIXME: These probably should use a getter method
```

Comment:
Is there a reason this FIXME isn't being addressed now? The current code would be against our style guide.